### ISSUE TITLE
Perform proper checks for constrained TypeVar when used within a generic class or function

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#1200]
When a constrained TypeVar (one with more than one type constraint) is used within a generic class or function, it is currently treated as an Any type.

For example, the following code should generate an error, but it doesn't.

```python
_T1 = TypeVar("_T1", int, Dict[str, str])

def func(a: _T1, b: _T1):
    return a + b
```

**Comment by erictraut** [comment]
This is now addressed in pyright 1.1.90, which I just published. It will also be included in the next release of pylance.
